### ISSUE TITLE
Validate Rule Format

### ISSUE THREAD

**Issue by carlosamoreno** [koorchik/js-validator-livr#30]
Hello there,

is there any way to validate a rule? I mean, if I create a rule that is not valid, only fails while validating, but not in the constructor. For instance,

```
const livr = require('livr');

const validator = new livr.Validator("I'm not a rule");
validator.validate({ prop: 'whatever' });
```

Returns the following error _Rule [I] not registered_ in the *validate* method

### Proposed Solution
Throw an error in the _Validator_ constructor. Invoking a _validateRule_ method before creating the _Validator_ object.
Another possible solution could be adding a public method for validating the rule. In this way, we have a way to validate the rule before creating the Validator.

**Comment by koorchik** [comment]
This should throw error too
```javascript
validator.prepare()
```
"prepare" will run all rule builders and will cache validation closures
